File: backend/src/main/java/com/medistock/inventory/service/InventoryService.java
package com.medistock.inventory.service;

import com.medistock.inventory.dto.request.InventoryRequest;
import com.medistock.inventory.dto.response.InventoryResponse;

import java.util.List;
import java.time.LocalDate;

public interface InventoryService {
    InventoryResponse createInventory(InventoryRequest request);
    List<InventoryResponse> getAllInventory();
    InventoryResponse getInventoryById(Long id);
    InventoryResponse updateInventory(Long id, InventoryRequest request);
    void deleteInventory(Long id);
    List<InventoryResponse> getLowStockInventory();
    List<InventoryResponse> getExpiredInventory();
    List<InventoryResponse> getExpiringInventory(int days);
    List<InventoryResponse> getInventoryByMedicineId(Long medicineId);
        InventoryResponse receivePurchaseOrderStock(
            Long medicineId,
            Integer quantity,
            Integer minimumStock,
            String batchNumber,
            LocalDate expiryDate,
            String storageLocation,
            String purchaseOrderNumber
    );
}

File: frontend/src/services/api/inventoryService.js
import apiClient from './apiClient';

export const inventoryService = {
  getAllInventory: async () => {
    const response = await apiClient.get('/inventory');
    return response.data;
  },

  getInventoryById: async (id) => {
    const response = await apiClient.get(`/inventory/${id}`);
    return response.data;
  },

  getLowStockInventory: async () => {
    const response = await apiClient.get('/inventory/low-stock');
    return response.data;
  },

  getExpiredInventory: async () => {
    const response = await apiClient.get('/inventory/expired');
    return response.data;
  },

  getExpiringInventory: async (days = 30) => {
    const response = await apiClient.get('/inventory/expiring', {
      params: { days },
    });
    return response.data;
  },

  getInventoryByMedicineId: async (medicineId) => {
    const response = await apiClient.get(`/inventory/medicine/${medicineId}`);
    return response.data;
  },

  createInventory: async (data) => {
    const response = await apiClient.post('/inventory', data);
    window.dispatchEvent(new Event('medistock-inventory-updated'));
    return response.data;
  },

  updateInventory: async (id, data) => {
    const response = await apiClient.put(`/inventory/${id}`, data);
    window.dispatchEvent(new Event('medistock-inventory-updated'));
    return response.data;
  },

  deleteInventory: async (id) => {
    const response = await apiClient.delete(`/inventory/${id}`);
    window.dispatchEvent(new Event('medistock-inventory-updated'));
    return response.data;
  },

  getStockMovements: async (type, search) => {
    const response = await apiClient.get('/inventory/movements', {
      params: { type, search },
    });
    return response.data;
  },
};
